libraries/entities/src/index.js

```diff
@@ -0,0 +1,633 @@
+        `You can't modify elements of the rowKey`
+      );
+      assert(
+        this._partitionKey === this.constructor.__partitionKey.exact(newProperties),
+        `You can't modify elements of the partitionKey`
+      );
+
+      try {
+        entity = this.constructor.serialize(newProperties);
+        [result] = await this._db.fns[`${this._tableName}_modify`](this._partitionKey, this._rowKey, entity, 1, this._etag);
+      } catch (e) {
+        if (e.code === 'P0004') {
+          return null;
+        }
+
+        if (e.code === 'P0002') {
+          const err = new Error('Resource not found');
+
+          err.code = 'ResourceNotFound';
+          err.statusCode = 404;
+
+          throw err;
+        }
+
+        throw e;
+      }
+
+      this._getPropertiesFromEntity(entity);
+      this._etag = result.etag;
+
+      return this;
+    };
+
+    let result;
+    while (attemptsLeft--) {
+      result = await attemptModify();
+
+      if (result) {
+        break;
+      }
+
+      await this.reload();
+    }
+
+    if (attemptsLeft <= 0) {
+      const err = new Error('MAX_MODIFY_ATTEMPTS exhausted, check for congestion');
+      err.code = 'EntityWriteCongestionError';
+      err.originalEntity = this._properties;
+      throw err;
+    }
+
+    return result;
+  }
+
+  static _getContextEntries(contextNames, contextEntries) {
+    const ctx = {};
+
+    assert(typeof contextEntries === 'object' && contextEntries.constructor === Object, 'context should be an object');
+
+    contextNames.forEach(key => {
+      assert(key in contextEntries, `context key ${key} must be specified`);
+    });
+
+    Object.entries(contextEntries).forEach(([key, value]) => {
+      assert(contextNames.includes(key), `context key ${key} was not declared in Entity.configure`);
+      ctx[key] = value;
+    });
+
+    return ctx;
+  }
+
+  static _doCondition(conditions, options) {
+    const valueFromOperand = (type, operand) => {
+      if (operand instanceof Date) {
+        return `'${operand.toJSON()}'`;
+      }
+
+      throw new Error('condition operand can only be a date');
+    };
+
+    if (!conditions) {
+      return null;
+    }
+
+    if (conditions) {
+      assert(typeof conditions === 'object' && conditions.constructor === Object, 'conditions should be an object');
+    }
+
+    let pk;
+    let rk;
+    let condition = [];
+    let covered = [];
+
+    try {
+      pk = this.__partitionKey.exactFromConditions(conditions);
+    } catch (e) {
+      if (options.matchPartition === 'exact') {
+        assert(pk, 'conditions should provide enough constraints for constructions of the partition key');
+      }
+    }
+
+    try {
+      rk = this.__rowKey.exactFromConditions(conditions);
+    } catch (e) {
+      if (options.matchRow === 'exact') {
+        assert(rk, 'conditions should provide enough constraints for constructions of the row key');
+      }
+    }
+
+    if (pk) {
+      condition.push(`partition_key = '${pk}'`);
```

Comment:
This (and `'${rk}'` below) is substituting an arbitrary value into SQL.  I think we'll need to change the stored function to take an `rk` and `pk` argument and use placeholders to pass those into the resulting SQL statement.

Reply:
Done!